src/Codeception/Lib/Console/Message.php

```diff
@@ -78,44 +95,41 @@ public function append($string)
         return $this;
     }
 
-    public function apply($func)
+    public function apply(callable $func): self
     {
         $this->message = call_user_func($func, $this->message);
         return $this;
     }
 
-    public function center($char)
+    public function center(string $char): self
     {
         $this->message = $char . $this->message . $char;
         return $this;
     }
 
-    /**
-     * @return mixed
-     */
-    public function getMessage()
+    public function getMessage(): string
     {
         return $this->message;
     }
 
-    public function block($style)
+    public function block(string $style): self
     {
         $this->message = $this->output->formatHelper->formatBlock($this->message, $style, true);
 
         return $this;
     }
 
-    public function getLength($includeTags = false)
+    public function getLength(bool $includeTags = false): int
     {
         return mb_strwidth($includeTags ? $this->message : strip_tags($this->message), 'utf-8');
     }
 
-    public static function ucfirst($text)
+    public static function ucfirst(?string $text): string
```

Comment:
Questionmark in signature and all the casting looks suspicious.
What code passes nulls to this method? I think that it should be fixed.

Reply:
The uses are here.. `src/Codeception/Test/Cept.php` : 

```php
    public function toString(): string
    {
        return $this->getSignature() . ': ' . Message::ucfirst($this->getFeature());
    }
```

... and here `src/Codeception/Test/Cest.php` :

```php
    public function toString(): string
    {
        return sprintf('%s: %s', ReflectionHelper::getClassShortName($this->getTestClass()), Message::ucfirst($this->getFeature()));
    }
```

Then, both the class `src/Codeception/Test/Feature/ScenarioLoader.php`:

```php
    public function getFeature(): ?string
    {
        return $this->getScenario()->getFeature();
    }
```

as the interface `src/Codeception/Test/Interfaces/ScenarioDriven.php`:

```php
    public function getFeature(): ?string;
```
    
have the return value nullable.

What should I do?

Reply:
@Naktibalda

Reply:
@Naktibalda also look at the failed build: https://github.com/Codeception/Codeception/actions/runs/617723034 .

Reply:
I think that it is Gherkin's fault that Feature can be null, and it would be the best to fix this problem at the source and add casting to string to https://github.com/Codeception/Codeception/blob/5.0/src/Codeception/Test/Gherkin.php#L63

Reply:
@Naktibalda Please review my last commit.
Also note that I will finish adding typehints to the files in `src/Codeception/*php` in the next PR.